packages/account/src/Sections/Security/ConnectedApps/connected-apps.jsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import { DesktopWrapper, MobileWrapper, Button, Modal, Icon, DataTable, DataList } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import ErrorComponent from 'Components/error-component';
+import { WS } from 'Services/ws-methods';
+import getConnectedAppsColumnsTemplate from './data-table-template';
+
+class ConnectedApps extends React.Component {
+    state = {
+        is_loading: true,
+        is_modal_open: false,
+        selected_app_id: null,
+        is_error: false,
+        connected_apps: [],
+    };
+    columns_map = getConnectedAppsColumnsTemplate(this.handleToggleModal).reduce((map, item) => {
+        map[item.col_index] = item;
+        return map;
+    }, {});
+
+    componentDidMount() {
+        this.updateApp();
+    }
+
+    updateApp = () => {
+        this.fetchConnectedApps()
+            .then((response) => {
+                this.setState({
+                    is_loading: false,
+                    connected_apps: response,
+                });
+            })
+            .catch(() => {
+                this.setState({ is_error: true });
+            });
+    };
+
+    handleRevokeAccess = () => {
+        this.setState({ is_modal_open: false, is_loading: true });
+        this.revokeConnectedApp(this.state.selected_app_id).then(this.updateApp);
+    };
+
+    handleToggleModal = (app_id = null) => {
+        if (this.state.is_modal_open) {
+            this.setState({ is_modal_open: false, selected_app_id: null });
+        } else {
+            this.setState({ is_modal_open: true, selected_app_id: app_id });
+        }
+    };
+
+    fetchConnectedApps = () => {
```

Comment:
same here: I think WS.send already returns a promise?

Reply:
fixed it.